typing/printtyp.ml

```diff
@@ -1527,49 +1527,35 @@ let extension_constructor_args_and_ret_type_subtree ext_args ext_ret_type =
   let args = tree_of_constructor_arguments ext_args in
   (args, ret)
 
-let isolated_extension_constructor_args_and_ret_type_subtree
-    ext_args ext_ret_type =
-  match ext_ret_type with
-  | None ->
-      extension_constructor_args_and_ret_type_subtree ext_args ext_ret_type
-  | Some _ ->
-      Names.with_local_names
-        (fun () ->
-           extension_constructor_args_and_ret_type_subtree
-             ext_args ext_ret_type
-        )
-
-(* When printing extension constructor, we need to forget the naming decisions
-   of different parts of the extension constructor depending on the context.
-   For example, in
-   {[
-     type 'b t += A: 'b -> 'b any t
-   ]}
-   We can choose to either use local names for the right hand GADT constructor
-   (Isolated_gadt), or the left hand type parameters (Isolated_param).
-   Isolating the gadt constructor is the same behaviour as
-   [isolated_tree_of_constructor].
-   We would choose to isolate the type parameters if we want to refer to the
-   GADT variables from another part of the printing context.
-   *)
-type extension_constructor_scoping =
-  | Isolated_gadt
-  | Isolated_param
-
-let add_extension_constructor_to_preparation scoping ext =
-  begin match scoping with
-  | Isolated_gadt ->
-    let ty_params = filter_params ext.ext_type_params in
-    List.iter add_alias ty_params;
-    List.iter prepare_type ty_params;
-    List.iter add_printed_alias ty_params;
-  | Isolated_param -> ()
-  end;
+(* When printing extension constructor, it is important to ensure that
+after printing the constructor, we are still in the scope of the constructor.
```

Comment:
I don't understand this sentence (also probably "when printing extension constructor[+s]"?).

What does "after printing the constructor" refer to here? Does 'the constructor' mean `A : ...` or `A of ...`, or just `A`?